### PR TITLE
Route tor dns requests to localhost:5400.

### PR DESCRIPTION
**Pull request by rtreffer**
We should never disclose the use of a .onion domain. All DNS
queries for a .onion domain are thus treated as tor requests.
They should generate an error afterwards, which means there is
room for optimization.

Every tor tagged acccount will query dns through tor.

### CODE REVIEW

**Comment by SamWhited** [comment]
A quick problem I see at first glance: Even if the option in settings is set to "Never" .onion's are still queried through Tor. I agree with you that they should generally be routed through Tor (eg. if "auto" is specified onioins should still always go through Tor), but if "Never" is specified, that should be a cue that the user wanted to specifically override that behavior (no idea why they'd want to do that though).

TL;DR — right now "only .onion" and "never" options are the same thing.

**Comment by rtreffer** [comment]
Well, Never with .onion shouldn't even connect. There is no server that could handle .onion, so how would you connect in that case?

**Comment by SamWhited** [comment]
yah, it doesn't really make sense, I just figure if the uesr sets it to "Never" for some reason, they really mean "dont' use Tor ever".

Maybe just removing "never" would be a better solution (so "only .onion" essentially becomes the "Don't use Tor" option)

**Comment by rtreffer** [comment]
Ok, so typing an .onion into the account box results in the correct dns being used.

D/conversations(30964): using dns server: 127.0.0.1 to look up okj7xc6j2szr2y75.onion
D/conversations(30964): artyonion@okj7xc6j2szr2y75.onion: error connecting account. try again in 189s for the 6 time

Obviously this is correct, as there is no way to really resolve .onion.

**Comment by SamWhited** [comment]
@rtreffer Awesome; thanks! I'll go ahead and merge this and we can sort out any issues upstream.

**Comment by rtreffer** [comment]
Trying to use a normal account... Still "connecting", but the DNS part works.

**Comment by rtreffer** [comment]
Or not, not 100% sure.... It takes forever.....

**Comment by rtreffer** [comment]
Ok, we need a different DNS timeout.... This is currently rather low on.
